test/intervals/test_intervals.jl

```diff
@@ -0,0 +1,239 @@
+function simple_intersection(intervals_a, intervals_b)
+    sort!(intervals_a)
+    sort!(intervals_b)
+
+    intersections = Any[]
+
+    i = 1
+    j = 1
+    while i <= length(intervals_a) && j <= length(intervals_b)
+        ai = intervals_a[i]
+        bj = intervals_b[j]
+
+        if Intervals.alphanum_isless(ai.seqname, bj.seqname) ||
+           (ai.seqname == bj.seqname && ai.last < bj.first)
+            i += 1
+        elseif Intervals.alphanum_isless(bj.seqname, ai.seqname) ||
+               (ai.seqname == bj.seqname && bj.last < ai.first)
+            j += 1
+        else
+            k = j
+            while k <= length(intervals_b) && intervals_b[k].first <= ai.last
+                if isoverlapping(ai, intervals_b[k])
+                    push!(intersections, (ai, intervals_b[k]))
+                end
+                k += 1
+            end
+            i += 1
+        end
+    end
+
+    return intersections
+end
+
+
+facts("IntervalCollection") do
+
+    context("Insertion/Iteration") do
+        n = 100000
+        intervals = random_intervals(["one", "two", "three"], 1000000, n)
+        ic = IntervalCollection{Int}()
+
+        @fact isempty(ic) => true
+        @fact collect(Interval{Int}, ic) == Interval{Int}[] => true
+
+        for interval in intervals
+            push!(ic, interval)
+        end
+        @fact collect(ic) == sort(intervals) => true
+    end
+
+
+    context("Intersection") do
+        n = 1000
+        srand(1234)
+        intervals_a = random_intervals(["one", "two", "three"], 1000000, n)
+        intervals_b = random_intervals(["one", "three", "four"], 1000000, n)
+
+        # empty versus empty
+        ic_a = IntervalCollection{Int}()
+        ic_b = IntervalCollection{Int}()
+        @fact collect(intersect(ic_a, ic_b)) == Any[] => true
+
+        # empty versus non-empty
+        for interval in intervals_a
+            push!(ic_a, interval)
+        end
+
+        @fact collect(intersect(ic_a, ic_b)) == Any[] => true
+        @fact collect(intersect(ic_b, ic_a)) == Any[] => true
+
+        # non-empty versus non-empty
+        for interval in intervals_b
+            push!(ic_b, interval)
+        end
+
+        @fact sort(collect(intersect(ic_a, ic_b))) ==
+              sort(simple_intersection(intervals_a, intervals_b)) => true
+    end
+
+
+    context("Show") do
+        nullout = open("/dev/null", "w")
+
+        ic = IntervalCollection{Int}()
+        show(nullout, ic)
+
+        push!(ic, Interval{Int}("one", 1, 1000, STRAND_POS, 0))
+        show(nullout, ic)
+
+        intervals = random_intervals(["one", "two", "three"], 1000000, 100)
+        for interval in intervals
+            push!(ic, interval)
+        end
+        show(nullout, ic)
+
+        show(nullout, STRAND_NA)
+        show(nullout, STRAND_POS)
+        show(nullout, STRAND_NEG)
+        show(nullout, STRAND_BOTH)
+    end
+end
+
+
+facts("Alphanumeric Sorting") do
```

Comment:
This should come before `facts("IntervalCollection")`, since the functions tested here are used tests in that block.